test/simulator/test/org/apache/cassandra/simulator/test/EpochStressTest.java

```diff
@@ -0,0 +1,231 @@
+ --add-opens java.base/jdk.internal.loader=ALL-UNNAMED
+ --add-opens java.base/jdk.internal.ref=ALL-UNNAMED
+ --add-opens java.base/jdk.internal.reflect=ALL-UNNAMED
+ --add-opens java.base/jdk.internal.math=ALL-UNNAMED
+ --add-opens java.base/jdk.internal.module=ALL-UNNAMED
+ --add-opens java.base/jdk.internal.util.jar=ALL-UNNAMED
+ --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED
+ --add-opens jdk.management.jfr/jdk.management.jfr=ALL-UNNAMED
+ --add-opens java.desktop/com.sun.beans.introspect=ALL-UNNAMED
+ */
+public class EpochStressTest extends SimulationTestBase
+{
+    @Test
+    public void manyEpochsAndAccordConverges() throws IOException
+    {
+        simulate(simulation -> {
+                     // setup
+                     ClusterActions.Options options = noActions(simulation.cluster.size());
+                     ClusterActions clusterActions = new ClusterActions(simulation.simulated, simulation.cluster,
+                                                                        options, new ClusterActionListener.NoOpListener(), new Debug(new EnumMap<>(Debug.Info.class), new int[0]));
+                     return ActionList.of(clusterActions.initializeCluster(initializeAll(simulation.cluster.size())),
+                                          simulation.schemaChange(1, "CREATE KEYSPACE ks WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor' : 3}"),
+                                          simulation.schemaChange(1, "CREATE TABLE IF NOT EXISTS ks.tbl (pk int PRIMARY KEY, v int) WITH " + TransactionalMode.full.asCqlParam()));
+                 },
+                 simulation -> {
+                     // test
+                     RandomSource random = simulation.simulated.random;
+                     int numEpochs = 100;
+                     List<Action> actions = new ArrayList<>(numEpochs);
+                     for (int i = 0; i < numEpochs; i++)
+                     {
+                         int node = random.uniform(1, simulation.cluster.size() + 1);
+                         actions.add(simulation.schemaChange(node, "ALTER TABLE ks.tbl WITH comment = 'step=" + i + "'"));
+                     }
+                     return ActionList.of(actions);
+                 },
+                 simulation -> {
+                     // teardown
+                     List<Action> actions = new ArrayList<>(simulation.cluster.size());
+                     for (int i = 0; i < simulation.cluster.size(); i++)
+                         actions.add(HarrySimulatorTest.lazy(simulation.simulated, simulation.cluster.get(i + 1), EpochStressTest::validate));
+                     return ActionList.of(actions);
+                 },
+                 config -> config.nodes(3, 3)
+                                 .dcs(1, 1)
+                                 .threadCount(100));
+    }
+
+    private static void validate()
+    {
+        Logger logger = LoggerFactory.getLogger(EpochStressTest.class);
```

Comment:
I don't know why but having it in static seemed to loose the logs, but putting it here I got the logs... I have no clue why....

Reply:
I do see several runs with no logs, this worries me...